File: core/src/main/java/net/m3mobile/core/requester/FinishRequiredBroadcastRequester.kt
package net.m3mobile.core.requester

import android.content.Intent
import net.m3mobile.core.InternalM3Api

/**
 * 본래 원하는 요청을 보낸 후, 설정 완료 신호를 추가로 보내야 하는 브로드캐스트 전용 클래스입니다.
 */
@InternalM3Api
public abstract class FinishRequiredBroadcastRequester: BroadcastRequester() {

    override fun request() {
        super.request()

        val finishIntent = Intent("com.android.server.startupservice.config.fin")
        context.sendBroadcast(finishIntent)
    }
}

File: dotnet/M3Mobile.M3Sdk.Xamarin/Internal/FinishRequiredBroadcastRequester.cs
using Android.Content;

namespace M3Sdk.Xamarin.Internal
{
    internal abstract class FinishRequiredBroadcastRequester : BroadcastRequester
    {
        protected FinishRequiredBroadcastRequester(Context context)
            : base(context)
        {
        }

        /// <summary>
        /// Sends the configured request broadcast and then the StartUp finish notification broadcast.
        /// </summary>
        internal override void Request()
        {
            base.Request();

            var finishIntent = new Intent(Constants.StartUp.RequestConfigFinish);
            Context.SendBroadcast(finishIntent);
        }
    }
}
